core/execution/execution.h

```diff
@@ -20,18 +20,19 @@ struct TRTEngine : torch::CustomClassHolder {
     std::pair<uint64_t, uint64_t> num_io;
     EngineID id;
     std::string name;
+    std::string device_info;
```

Comment:
Why do we hold the device info string instead of the struct?

Reply:
This is serialized device struct. Since, we don't use it directly as a struct anywhere else other than serialization. It might save us time in directly using it at def_pickle

Reply:
It should just be a function call in the def_pickle function, I think its a bit odd to hold a serialized version of the device info in the deserialized object

Reply:
Changed to CudaDevice struct instead